### PR TITLE
[Gallery] Add Material Sliders demo

### PR DESCRIPTION
**Pull request by perclasson**
Add Material Sliders demo.

Simulator Screen Shot - iPhone 11 Pro Max - 2019-12-20 at 13 10 26
Simulator Screen Shot - iPhone 11 Pro Max - 2019-12-20 at 17 24 33
Simulator Screen Shot - iPhone 11 Pro Max - 2019-12-20 at 17 24 31
Simulator Screen Shot - iPhone 11 Pro Max - 2019-12-20 at 17 24 28

Closes https://github.com/material-components/material-components-flutter-gallery/issues/188

### CODE REVIEW

**Comment by clocksmith** [comment]
Like we did for Cupertino slider demo, is spaceEvenly better for the main column axis alignment?

**Comment by perclasson** [comment]
> Like we did for Cupertino slider demo, is spaceEvenly better for the main column axis alignment?

When it has spaceEvenly it looks a bit too sparse in the app. That is why I decided to go with a SixedBox in between instead. Here's how spaceEvenly looks like:

Simulator Screen Shot - iPhone 11 Pro Max - 2019-12-20 at 14 50 47

File: gallery/gallery/lib/demos/material/sliders_demo.dart [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+        sliders = _CustomSliders();
+    }
+    return Scaffold(
+      appBar: AppBar(
+        automaticallyImplyLeading: false,
+        title: Text(_title(context)),
+      ),
+      body: Theme(
+        // Set platform to display material sliders only.
+        data: Theme.of(context).copyWith(platform: TargetPlatform.android),
```
Comment (rami-a) [inline gallery/gallery/lib/demos/material/sliders_demo.dart]: Why is this necessary? Oh I see it uses `Slider.adaptive`. I'd remove this `Theme` override and replace the `Slider.adaptive` calls with regular `Slider`

File: gallery/gallery/lib/demos/material/sliders_demo.dart [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+                  width: 64,
+                  height: 48,
+                  child: TextField(
+                    textAlign: TextAlign.center,
+                    onSubmitted: (value) {
+                      final double newValue = double.tryParse(value);
+                      if (newValue != null && newValue != _continuousValue) {
+                        setState(() {
+                          _continuousValue =
+                              newValue.clamp(0.0, 100.0) as double;
```
Comment (rami-a) [inline gallery/gallery/lib/demos/material/sliders_demo.dart]: If newValue is of type double, why do you need to cast here?

File: gallery/gallery/lib/demos/material/sliders_demo.dart [hunk 3]
```diff
@@ -0,0 +1,504 @@
...
+}
+
+class _RangeSlidersState extends State<_RangeSliders> {
+  RangeValues _continuousValues = const RangeValues(25.0, 75.0);
+  RangeValues _discreteValues = const RangeValues(40.0, 120.0);
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 40.0),
```
Comment (rami-a) [inline gallery/gallery/lib/demos/material/sliders_demo.dart]: Drop the `.0` everywhere you can in this file

File: gallery/gallery/lib/demos/material/sliders_demo.dart [hunk 4]
```diff
@@ -0,0 +1,504 @@
...
+            thumbPath = _leftTriangle(size, center);
+            break;
+          case Thumb.end:
+            thumbPath = _rightTriangle(size, center);
+            break;
+        }
+        break;
+    }
+    canvas.drawPath(
+        thumbPath, Paint()..color = colorTween.evaluate(enableAnimation));
```
Comment (rami-a) [inline gallery/gallery/lib/demos/material/sliders_demo.dart]: Trailing comma for formatting

File: gallery/gallery/lib/demos/material/sliders_demo.dart [hunk 5]
```diff
@@ -0,0 +1,504 @@
...
+  }) {
+    final canvas = context.canvas;
+    final colorTween = ColorTween(
+      begin: sliderTheme.disabledThumbColor,
+      end: sliderTheme.thumbColor,
+    );
+    final size = _thumbSize * sizeTween.evaluate(enableAnimation);
+    final thumbPath = _downTriangle(size, thumbCenter);
+    canvas.drawPath(
+        thumbPath, Paint()..color = colorTween.evaluate(enableAnimation));
```
Comment (rami-a) [inline gallery/gallery/lib/demos/material/sliders_demo.dart]: Trailing comma

File: gallery/gallery/lib/demos/material/sliders_demo.dart (hunk 1 above)
Comment (perclasson) [reply gallery/gallery/lib/demos/material/sliders_demo.dart]: Good call.

File: gallery/gallery/lib/demos/material/sliders_demo.dart (hunk 2 above)
Comment (perclasson) [reply gallery/gallery/lib/demos/material/sliders_demo.dart]: Seems to be a dart issue:
https://github.com/dart-lang/sdk/issues/39652

File: gallery/gallery/lib/l10n/gallery_localizations.dart [hunk 6]
```diff
@@ -125,6 +124,7 @@ class GalleryLocalizations {
     Locale('en', 'IN'),
     Locale('en', 'NZ'),
     Locale('en', 'SG'),
+    Locale('en', 'US'),
```
[the remark this answers is not in the document]
Comment (perclasson) [reply gallery/gallery/lib/l10n/gallery_localizations.dart]: Updated.

[the remark this answers is not in the document]
Comment (shihaohong) [reply gallery/gallery/lib/l10n/gallery_localizations.dart]: Yes, I've been meaning to add this functionality but haven't had the opportunity to so far. I may work on it today!
